src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -50,9 +50,18 @@ const (
 	sessionIDLength   = 24
 )
 
+type sessionsDict map[string][]authtypes.LoginType
+
+func (d sessionsDict) Get(key string) []authtypes.LoginType {
```

Comment:
As far as I know there's no way to override an operator like `[]`, so this is the next best option if we're going for overloading. The other option is just to test if the map isn't initialized yet and do so if not, but this feels cleaner overall.